server/src/lib.rs

```diff
@@ -435,6 +441,14 @@ impl ServerConn {
             }
         ) {
             Ok(stream) => {
+                if self.streams.is_empty() {
```

Comment:
is_empty should be !has_available per discussion on IRC

Reply:
Yup. Will fix.